File: IrrgartenRuby/modelo/shield.rb
# frozen_string_literal: true
require_relative 'dice'
module Modelo
  class Shield
    def initialize(protection, uses)
      @protection = protection
      @uses = uses
    end
    def attack
      if @uses > 0
        @uses -= @uses
        return @protection
      end
      0.0
    end

    def to_s
      "S [#{@protection} , #{@uses} ]"
    end

    def discard()
      Dice.discard_element(@uses)
    end


  end
end

File: IrrgartenJava/src/modelo/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package modelo;

/**
 *
 * @author nmatl
 */
public class Shield {
    private float protection;
    private int uses;
    
    public Shield(float protection, int uses){
        this.protection = protection;
        this.uses = uses;
    }
    
    public float protect(){
        if(this.uses > 0){
            this.uses--;
            return this.protection;
        }
        return 0;
    }
    
    public String toString(){
        return "S[" + this.protection + "," + this.uses + "]";
    }
    public boolean discard(){
        return Dice.discardElement(uses);
    }
    
}
